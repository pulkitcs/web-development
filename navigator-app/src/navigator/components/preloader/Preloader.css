.preloader {
  height: 100%;
  width: 100%;
  position: absolute;
  background-color: rgba(255, 255, 255, .7);
  top: 0;
  left: 0;
}

.preloader4 {
  position: absolute;
  top: 25%;
  left: 50%;
  margin-left: -18px;
  width:35px;
  height:35px;
  display:inline-block;
  padding:0px;
  border-radius:100%;
  border:2px solid;
  border-top-color:rgba(0,0,0, 0.65);
  border-bottom-color:rgba(0,0,0, 0.15);
  border-left-color:rgba(0,0,0, 0.65);
  border-right-color:rgba(0,0,0, 0.15);
  -webkit-animation: preloader4 0.8s linear infinite;
  animation: preloader4 0.8s linear infinite;
}
@keyframes preloader4 {
  from {transform: rotate(0deg);}
  to {transform: rotate(360deg);}
}
@-webkit-keyframes preloader4 {
  from {-webkit-transform: rotate(0deg);}
  to {-webkit-transform: rotate(360deg);}
}