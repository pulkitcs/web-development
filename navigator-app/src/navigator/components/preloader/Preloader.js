import React, { PureComponent } from 'react';
import './Preloader.css';
import PropTypes from 'prop-types';

class Preloader extends PureComponent {
  render() {
    const { enable } = this.props;
    return (
      enable && (
        <div className="preloader">
          <div className="preloader4"></div>
        </div>
      )
    )
  }
}

export default Preloader;

Preloader.propTypes = {
  enable: PropTypes.bool.isRequired
};
