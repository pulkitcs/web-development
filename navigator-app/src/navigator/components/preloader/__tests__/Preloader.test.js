import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Preloader from '../preloader';

describe('<Preloader /> component', () => {

  it('should match the snapshot', () => {
    const tree = renderer.create(<Preloader enable={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the HTML', () => {
    const enable = true; // Renders
    const wrapper = shallow(<Preloader enable={enable} />);
    
    expect(wrapper.find('.preloader').length).toBe(1); 
  })

  it('should not render the HTML', () => {
    const enable = false; // Does not render 
    const wrapper = shallow(<Preloader enable={enable} />);
    
    expect(wrapper.find('.preloader').length).toBe(0);
  })
})
