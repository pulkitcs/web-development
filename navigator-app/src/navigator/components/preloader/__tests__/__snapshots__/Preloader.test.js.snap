// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader /> component should match the snapshot 1`] = `
<div
  className="preloader"
>
  <div
    className="preloader4"
  />
</div>
`;
